### DISCUSSION TITLE
Is Azure AD interactive authentication supported on mobile?

### DISCUSSION THREAD

**Discussion by thenextman** [dotnet/SqlClient#3795] [Q&A]
Running on .NET 6 on an iOS system and attempting to use interactive Azure authentication seems like it can never work.

The browser is launched and it's possible to authenticate, but then attempts a redirect to localhost (which can never succeed). The only way out is to cancel the authentication, making the connection fail.

On NET Core, the redirect URL is always set to localhost:

https://github.com/dotnet/SqlClient/blame/2b31810ce69b88d707450e2059ee8fbde63f774f/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ActiveDirectoryAuthenticationProvider.cs#L200

However, this cannot work on mobile (as per the underlying Azure.Identity / MSAL implementations). Is this supported or supposed to work somehow? It's not clear.

**Comment by Kaur-Parminder** [comment]
@thenextman  Sorry for delayed response.  I believe MSAL suppose to work as per documentation here https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Acquiring-tokens-interactively. 
and using localhost is prbably expected:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/MSAL.NET-uses-web-browser#at-a-glance
Sqlclient is doing same:
https://github.com/dotnet/SqlClient/blob/2b31810ce69b88d707450e2059ee8fbde63f774f/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ActiveDirectoryAuthenticationProvider.cs#L393

I had issues while trying to Repro it with browser pop-up. Can you provide simple App that reproduces it?

**Comment by maxbrousseau** [comment]
Hello @Kaur-Parminder ,

For a test app to reproduce the issue, you can check the project that I have linked here: https://github.com/dotnet/SqlClient/issues/2062#issuecomment-1602877151

Thank you,
